incompatibility/incompatibility-report/incomp-scanning.go

```diff
@@ -0,0 +1,36 @@
+// Copyright 2019 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package incompatibility
+
+import (
+	openapiv3 "github.com/googleapis/gnostic/openapiv3"
+	plugins "github.com/googleapis/gnostic/plugins"
+)
+
+// Create a report of incompatibilities, write protobuf message to
+// environment, bool indicates if detailed incompatibility report is
+// desired.
+func CreateIncompReport(env *plugins.Environment, detailed bool) {
```

Comment:
Can you pass an enum instead of a boolean?

Passing booleans around is not really considered best practice.

[Source](https://alexkondov.com/should-you-pass-boolean-to-functions/#:~:text=Flags%20are%20not%20a%20best,based%20on%20the%20call%20site.&text=Single%20argument%20functions%20and%20passing,named%20values%20improves%20the%20readability.)

Reply:
added enum to specify report type